free-monads/src/main/java/com/github/jorgecastillo/kotlinandroid/free/interpreter/Interpreter.kt

```diff
@@ -20,108 +20,108 @@ import com.github.jorgecastillo.kotlinandroid.domain.model.CharacterError.Authen
 import com.github.jorgecastillo.kotlinandroid.domain.model.CharacterError.NotFoundError
 import com.github.jorgecastillo.kotlinandroid.domain.model.CharacterError.UnknownServerError
 import com.github.jorgecastillo.kotlinandroid.free.algebra.HeroesAlgebra
-import com.github.jorgecastillo.kotlinandroid.free.algebra.HeroesAlgebraHK
-import com.github.jorgecastillo.kotlinandroid.free.algebra.HeroesAlgebraKind
-import com.github.jorgecastillo.kotlinandroid.free.algebra.ev
 import com.github.jorgecastillo.kotlinandroid.view.viewmodel.SuperHeroViewModel
 import com.karumi.marvelapiclient.model.CharacterDto
 import com.karumi.marvelapiclient.model.CharactersQuery.Builder
 import com.karumi.marvelapiclient.model.MarvelImage
 import kotlinx.coroutines.experimental.CommonPool
 import kotlinx.coroutines.experimental.async
+import com.github.jorgecastillo.kotlinandroid.free.algebra.*
 
 @Suppress("UNCHECKED_CAST")
 fun <F> interpreter(ctx: SuperHeroesContext, ME: MonadError<F, Throwable>,
-    AC: Async<F>): FunctionK<HeroesAlgebraHK, F> =
-    object : FunctionK<HeroesAlgebraHK, F> {
-      override fun <A> invoke(fa: HeroesAlgebraKind<A>): HK<F, A> {
-        val op = fa.ev()
-        return when (op) {
-          is HeroesAlgebra.GetAll -> getAllHeroesImpl(ctx, ME, AC)
-          is HeroesAlgebra.GetSingle -> getHeroDetailsImpl(ctx, ME, AC, op.heroId)
-          is HeroesAlgebra.HandlePresentationEffects -> {
-            ME.catch({ handlePresentationEffectsImpl(ctx, op.result) })
-          }
-          is HeroesAlgebra.Attempt<*> -> {
-            val result = op.fa.foldMap(interpreter(ctx, ME, AC), ME)
-            ME.attempt(result)
-          }
-        } as HK<F, A>
-      }
-    }
+                    AC: Async<F>): FunctionK<ForHeroesAlgebra, F> =
+        object : FunctionK<ForHeroesAlgebra, F> {
+            override fun <A> invoke(fa: HeroesAlgebraOf<A>): Kind<F, A> {
+                val op = fa.extract()
```

Comment:
thought we had moved on for `reify()`

Reply:
I believe it's going to be just `fix` before 0.7.0 @pakoito can clarify.

Reply:
It's going to be `fix`. Elizarov suggested not using `reify()` and the `extract()` it's too long to write.